packages/app/pages/course/[cid]/invite.tsx

```diff
@@ -1,77 +1,156 @@
-              marginTop: 200,
-            }}
-          >
-            <Card
-              style={{ width: 600, textAlign: "center" }}
-              cover={
-                <img
-                  alt="example"
-                  height="200"
-                  style={{ objectFit: "cover" }}
-                  src="https://open-2021.sites.olt.ubc.ca/files/2020/10/OSIP-2020-Slider.jpg"
-                />
-              }
-            >
-              <Space
-                direction="vertical"
-                style={{ width: "100%" }}
-                size="large"
-              >
-                <Meta title={cardMetaTitle} description={cardMetaDescription} />
-                <Button
-                  type="primary"
-                  style={{ width: "100%", height: 50, borderRadius: "5px" }}
-                >
-                  Accept Invitation
-                </Button>
-              </Space>
-            </Card>
-          </div>
-        </StandardPageContainer>
-      </>
-    );
-  } else if (course && code && code !== "demoCodeKey") {
+    const fetchData = async () => {
+      try {
+        const response = await API.course.getLimitedCourseResponse(
+          Number(cid),
+          String(code)
+        );
+        setCourse(response);
+      } catch (error) {
+        setCourseError(error);
+      }
+    };
+    if (cid) {
+      fetchData();
+    }
+  }, [cid, profile]);
+
+  // Register student and redirect to course
+  const addStudent = async (userData: UBCOuserParam) => {
+    await API.signup.registerStudent(userData);
+    localStorage.removeItem("lastVisited");
+    router.push("/courses");
+  };
+
+  // If the user is already in the course, redirect to courses page
+  if (
+    profile &&
+    profile.courses.some((userCourse) => userCourse.course.id === Number(cid))
+  ) {
+    localStorage.removeItem("lastVisited");
+    router.push("/courses");
+  }
+
+  const renderCard = (
+    title,
+    description,
+    buttonLabel,
+    buttonAction,
+    cover = null
+  ) => (
+    <Card style={{ width: 600, textAlign: "center" }} cover={cover}>
+      <h1>{title}</h1>
+      <Space direction="vertical" style={{ width: "100%" }} size="large">
+        <Meta title={description.title} description={description.text} />
+        <Button
+          type="primary"
+          style={{ width: "100%", height: 50, borderRadius: "5px" }}
+          onClick={buttonAction}
+        >
+          {buttonLabel}
+        </Button>
+      </Space>
+    </Card>
+  );
+
+  if (isLoading) {
     return (
-      <>
-        <StandardPageContainer>
-          <Head>
-            <title>Page Not Found or is it?</title>
-          </Head>
-          page not found?
-        </StandardPageContainer>
-      </>
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100vh",
+        }}
+      >
+        <Spin size="large" />
```

Comment:
Can we keep `tip`? 😅

Reply:
I don't know what you mean

Reply:
See documentation here https://4x.ant.design/components/spin/ abpit `tip`